### PR TITLE
Add Classes

### PR DESCRIPTION
**Pull request by Mujeeb4582**
# Task completed
- Create the classes
    - Person 
    - Student
    - Teacher
- Add the linter setup for ruby
- Update the readme file according to requirment

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(classroom, age, name = 'Unknown')
+    super(age, name)
+    @classroom = classroom
+  end
+
+  def play_hocky
```
Comment (Ghiftee) [inline student.rb]: - [ ] According to the requirements, the name of this method should be ```play_hooky``` Kindly correct it.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  private
+
+  def of_age?
```
Comment (Ghiftee) [inline person.rb:14]: - [ ] According to the requirements, the name of this method should be ```is_of_age``` Kindly correct it.

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require './person'
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(specialization, age, name = 'Unknown')
+    super(age, name)
```
Comment (Ghiftee) [inline teacher.rb]: - [ ] Kindly ensure that you are also checking for ```parent permission``` here

```
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+require './person'
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(classroom, age, name = 'Unknown')
+    super(age, name)
```
Comment (Ghiftee) [inline student.rb]: - [ ] Kindly ensure you are also checking for ```parent_permission``` here.
```ruby
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
```

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @Mujeeb4582  👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Classes were created
✔️ Associations were correctly created
✔️ README is professional

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: teacher.rb (hunk 3 above)
Comment (Mujeeb4582) [reply teacher.rb]: Dear @Ghiftee I think there is no need for `parent permission` in `teacher class`. Please read the requirements and give me feedback. I am waiting for your response.

File: person.rb (hunk 2 above)
Comment (Mujeeb4582) [reply person.rb:14]: @Ghiftee i corrected to `is_of_age?` but linter's give me error and give me suggestion make it `of_age?`, that why I again change it `of_age?`.

**Review by frank1738 — APPROVED** [review]
Hi @Mujeeb4582 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**